src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -921,4 +923,69 @@ class DeltaSuite extends QueryTest
       checkAnswer(spark.table("sc15200test"), Seq.empty)
     }
   }
+
+  test("support for setting dataChange to false") {
+    withTempDir { tempDir =>
+      if (tempDir.exists()) {
+        assert(tempDir.delete())
+      }
+
+      spark.range(100)
+        .repartition(4)
+        .write
+        .format("delta")
+        .save(tempDir.toString)
+
+      val df = spark.read.format("delta").load(tempDir.toString)
+
+      df
+        .repartition(2)
+        .write
+        .format("delta")
+        .mode("overwrite")
+        .option("dataChange", "false")
+        .save(tempDir.toString)
+
+      val deltaLog = DeltaLog.forTable(spark, tempDir)
+      val version = deltaLog.snapshot.version
+      val commitActions = deltaLog.store.read(FileNames.deltaFile(deltaLog.logPath, version))
+        .map(Action.fromJson)
+      val fileActions = commitActions.collect { case a: FileAction => a }
+      val dataChanges = fileActions.map(_.dataChange)
+
+      assert(dataChanges == Seq(false, false, false, false, false, false))
```

Comment:
Optional: If you want, you can write this (and the below one) as
`assert(fileActions.forAll(!_.dataChange))`